File: Data Structures & Algorithms/reorder-linked-list/submission-5.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    void reorderList(ListNode* head) {
        if(!head){
            return;
        }
        vector<ListNode*> nodes;
        ListNode* curr = head;
        while(curr){
            nodes.push_back(curr);
            curr=curr->next;
        }
        int i=0;
        int j = nodes.size()-1;
        while(i<j){
            ListNode* temp = nodes[i]->next;
            nodes[i]->next = nodes[j];
            nodes[j]->next = temp;
            i++;
            j--;
        }
        nodes[i]->next = nullptr;

    }
};

File: Data Structures & Algorithms/add-two-numbers/submission-5.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        multiplier = 1
        p = 0
        curr = l1
        while curr:
            p += multiplier*curr.val
            multiplier*=10
            curr = curr.next
        multiplier = 1
        pTwo = 0
        curr = l2
        while curr:
            pTwo += multiplier*curr.val
            multiplier*=10
            curr = curr.next
        res = str(p+pTwo)
        head = ListNode(0)
        curr = head
        for c in res[::-1]:
            curr.next = ListNode(int(c))
            curr = curr.next
        return head.next
        